### ISSUE TITLE
[driver-loader]: simplifying the logic behind the driver-loader

### ISSUE THREAD

**Issue by alacuku** [falcosecurity/charts#418]
**Motivation**
The `driver-loader` is an init-container that is deployed by default when a driver (module/ebpf) is used. Users can override the default configuration and disable the `driver-loader`. There are two config variables used to disable the driver loader:
https://github.com/falcosecurity/charts/blob/06225b85abae5beb77e1a11cf0d393a8b1bb8ce8/falco/values.yaml#L188-L193
The two variables are used to configure the deployment such in this case:
https://github.com/falcosecurity/charts/blob/06225b85abae5beb77e1a11cf0d393a8b1bb8ce8/falco/templates/pod-template.tpl#L133-L146
It is not enough to just set the `driver.loader.enabled=false` in order to have the volumes mounted in the falco pod but we need to set the `driver.loader.initContainer.enabled=false`. 
I see that the `driver-loader` logic is decoupled from the `initContainer` since we can have the former `enabled` and the latter `disabled`. This allows the users to provide their own logic for the `driver-loader` but for the end user, it's confusing and error-prone. The helm chart will deploy Falco but it will not start. It's not straightforward to debug such cases.

**Feature**
I think we should simplify the `driver-loader` logic in our helm chart by setting just one configuration variable that enables or disables it. We should implement some `helpers` that mount the right volumes in the right containers based on the configuration. If we really want we can still allow people to provide their own logic for the driver loader by asking them to explicitly configure the `init containers`, `volumes`, and `volumes mounts`. 
We should aim to offer a functional and straightforward experience when deploying Falco by preventing unnecessary over-engineering of the charts.

**Comment by Andreagit97** [comment]
Completely agree with that, having 2 flags is really confusing IMHO

**Comment by alacuku** [comment]
cc @leogr

**Comment by leogr** [comment]
Hey @alacuku 

Sorry, I lost track of this. Any updates?

**Comment by alacuku** [comment]
I was waiting for feedback from maintainers and the community.
 I would be happy to implement the proposed solution.

**Comment by leogr** [comment]
I agree the two settings are misleading, but in unifying them I see an issue:
```
loader: 
   # -- Enable/disable the init container. 
   enabled: true 
   initContainer: 
     # -- Enable/disable the init container. 
     enabled: true 
 ```
 
The first option is meant to enable the loader regardless of the init container. That is required because we have to support three different cases:

1. Driver loader via an init container: both options must be enabled
2. Driver loader in the same falco image (with no init container): the first option must be enabled, the `initContainer` must not
3. The user provides the driver by other means (eg. install it directly in the host): both options must be disabled, but the driver feature is still enabled

How can achieve this with only one setting? :thinking: 

PS 
The comment here :point_down: is wrong. That option should NOT enable the init container.
```
loader: 
   # -- Enable/disable the init container. 
   enabled: true 
```

**Comment by alacuku** [comment]
I would use always the init container, hence I will drop support for option two in the helm chart. That is the only case that requires both the switches: loader and initContainer. And the third options will still work since it does not require configuration from the chart side. 
Using an init container for the `falco-driver-loader` is in line with the best practices of kubernetes deployments.

**Comment by alacuku** [comment]
Currently the `falcosecurity/falco` does not work when used with the chart, please see here #419

**Comment by leogr** [comment]
> Currently the `falcosecurity/falco` does not work when used with the chart, please see here #419

See https://github.com/falcosecurity/charts/issues/419#issuecomment-1327522540 
Let's see if we want to remove that uses case :)
